### PR TITLE
Add the possibility to edit an equipment deletion.

### PR DESCRIPTION
**Pull request by flomillot**
Add the possibility to edit an equipment deletion.
Few code cleaning.

### CODE REVIEW

**Comment by etiennehomer** [comment]
Tests OK. 
There's no check on the applicability of the edited deletion. But we have this problem for all edition, it gonna be fixed in another US if needed

File: src/components/dialogs/equipment-deletion-dialog.js [hunk 1]
```diff
@@ -93,28 +90,26 @@ const EquipmentDeletionDialog = ({ open, onClose, currentNodeUuid }) => {
...
-        renderElement: (props) => (
-            <EquipmentItem
-                classes={equipmentClasses}
-                {...props}
-                key={props.element.key}
-                showsJustText={true}
-            />
-        ),
-    });
+    const [equipmentOrId, equipmentField, setEquipmentOrId] =
```
Comment (etiennehomer) [inline src/components/dialogs/equipment-deletion-dialog.js:93]: why equipmentOrId ?
equipmentId is fine no ?

File: src/components/graph/menus/modification-list-item.js [hunk 2]
```diff
@@ -84,13 +69,11 @@ export const ModificationListItem = ({
             return modif.attachedLineId;
         } else if (equipmentModificationModificationsType.has(modif.type)) {
             return modif.equipmentId;
-        } else if (useName && modif.equipmentName) {
```
Comment (etiennehomer) [inline src/components/graph/menus/modification-list-item.js:87]: Why remove this ? We never want the equipment name in this list ?

**Review by etiennehomer — APPROVED** [review]

File: src/components/graph/menus/modification-list-item.js (hunk 2 above)
Comment (etiennehomer) [reply src/components/graph/menus/modification-list-item.js:87]: Asked by PO, so ok with this removal

File: src/components/dialogs/equipment-deletion-dialog.js [hunk 3]
```diff
@@ -56,34 +54,26 @@ const makeItems = (eqpts, usesNames) => {
...
 
     const { snackError } = useSnackMessage();
 
     const intl = useIntl();
     const inputForm = useInputForm();
 
-    const equipmentClasses = useEquipmentStyles();
-
-    const [equipmentType, setEquipmentType] = useState('LINE');
+    const [equipmentType, setEquipmentType] = useState(editData?.equipmentType || 'LINE');
```
[the remark this answers is not in the document]
Comment (flomillot) [reply src/components/dialogs/equipment-deletion-dialog.js]: Ok you're right, I change it

File: src/components/graph/menus/modification-list-item.js [hunk 4]
```diff
@@ -84,13 +69,11 @@ export const ModificationListItem = ({
             return modif.attachedLineId;
         } else if (equipmentModificationModificationsType.has(modif.type)) {
```
[the remark this answers is not in the document]
Comment (flomillot) [reply src/components/graph/menus/modification-list-item.js]: Done

File: src/utils/rest-api.js [hunk 5]
```diff
@@ -2056,22 +2056,32 @@ export function deleteEquipment(
...
-    const deleteEquipmentUrl =
-        getStudyUrlWithNodeUuid(studyUuid, currentNodeUuid) +
-        '/network-modification/equipments/type/' +
-        encodeURIComponent(equipmentType) +
-        '/id/' +
-        encodeURIComponent(equipmentId);
-    return backendFetch(deleteEquipmentUrl, { method: 'delete' });
+    let deleteEquipmentUrl;
+    if (modificationUuid) {
+        console.info('Updating equipment deletion');
```
[the remark this answers is not in the document]
Comment (flomillot) [reply src/utils/rest-api.js:2064]: As said oraly I just standardized with the others rest api that create / edit a modification

**Review by laurgarn — APPROVED** [review]
Works OK.
Would need a more lazy loading when initialized value is not empty in autoCompleteField, in another US.

**Pull request merged**